### PR TITLE
Port of View-Switching Navigation sample to Unity

### PR DESCRIPTION
**Pull request by dvorn**
PR for #19 

Provided as modification of existing sample for ease of review.

If it's OK it can be made a new separate sample in another PR.

### CODE REVIEW

**Comment by dvorn** [comment]
@brianlagunas Should we create an umbrella folder for navigation projects, or just put Unity version beside MEF version?

**Comment by brianlagunas** [comment]
I haven't put any thoughts into the structure of the samples.  For now just stick it beside the MEF version.  When I get time I will be creating a standard for samples.

**Comment by dvorn** [comment]
@brianlagunas I understand that the port was approved.

File: View-Switching Navigation_Desktop/ViewSwitchingNavigation.Calendar/CalendarModule.cs [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
+        public CalendarModule(IUnityContainer unityContainer, IRegionManager regionManager)
+        {
+            this.unityContainer = unityContainer;
+            this.regionManager = regionManager;
+        }
 
         public void Initialize()
         {
+            this.unityContainer.RegisterType<ICalendarService, CalendarService>(new ContainerControlledLifetimeManager());
+            this.unityContainer.RegisterType<CalendarViewModel>(new ContainerControlledLifetimeManager());
```
Comment (brianlagunas) [inline View-Switching Navigation_Desktop/ViewSwitchingNavigation.Calendar/CalendarModule.cs]: Why are you registering Views and ViewModels with the container as singletons?  Don't do this.

File: View-Switching Navigation_Desktop/ViewSwitchingNavigation.Contacts/ContactsModule.cs [hunk 2]
```diff
@@ -1,30 +1,52 @@
...
+        {
+            this.unityContainer = unityContainer;
+            this.regionManager = regionManager;
+        }
 
         public void Initialize()
         {
+            this.unityContainer.RegisterType<IContactsService, ContactsService>(new ContainerControlledLifetimeManager());
+
+            this.unityContainer.RegisterType<ContactsViewModel>(new ContainerControlledLifetimeManager());
```
Comment (brianlagunas) [inline View-Switching Navigation_Desktop/ViewSwitchingNavigation.Contacts/ContactsModule.cs]: Don't register Views/ViewModels as singletons

File: View-Switching Navigation_Desktop/ViewSwitchingNavigation.Email/EmailModule.cs [hunk 3]
```diff
@@ -1,22 +1,38 @@
...
+            this.unityContainer = unityContainer;
+            this.regionManager = regionManager;
+        }
 
         public void Initialize()
         {
+            this.unityContainer.RegisterType<EmailNavigationItemView>(new ContainerControlledLifetimeManager());
+            this.unityContainer.RegisterType<IEmailService, EmailService>(new ContainerControlledLifetimeManager());
+
+            this.unityContainer.RegisterType<InboxView>(new ContainerControlledLifetimeManager());
```
Comment (brianlagunas) [inline View-Switching Navigation_Desktop/ViewSwitchingNavigation.Email/EmailModule.cs]: Don't register Views/ViewModels as singletons

File: View-Switching Navigation_Desktop/ViewSwitchingNavigation/QuickStartBootstrapper.cs [hunk 4]
```diff
@@ -1,22 +1,19 @@
...
-        private const string ModuleCatalogUri = "/ViewSwitchingNavigation;component/ModulesCatalog.xaml";
-
-        protected override void ConfigureAggregateCatalog()
+        protected override void ConfigureContainer()
         {
-            base.ConfigureAggregateCatalog();
+            base.ConfigureContainer();
 
-            this.AggregateCatalog.Catalogs.Add(new AssemblyCatalog(typeof(QuickStartBootstrapper).Assembly));
+            this.Container.RegisterType<Shell>(new ContainerControlledLifetimeManager());
```
Comment (brianlagunas) [inline View-Switching Navigation_Desktop/ViewSwitchingNavigation/QuickStartBootstrapper.cs]: Don't register Views/ViewModels as singletons

File: View-Switching Navigation_Desktop/ViewSwitchingNavigation/QuickStartBootstrapper.cs [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
 
 namespace ViewSwitchingNavigation
 {
-    public class QuickStartBootstrapper : MefBootstrapper
+    public class QuickStartBootstrapper : UnityBootstrapper
     {
-        private const string ModuleCatalogUri = "/ViewSwitchingNavigation;component/ModulesCatalog.xaml";
-
-        protected override void ConfigureAggregateCatalog()
+        protected override void ConfigureContainer()
```
Comment (brianlagunas) [inline View-Switching Navigation_Desktop/ViewSwitchingNavigation/QuickStartBootstrapper.cs]: This method can be removed

**Comment by dvorn** [comment]
@brianlagunas To simplify the job, I did formal translation MEF->Unity.  In MEF, they were singletons (the default).  In Unity the default is the opposite.  This way it's guaranteed to work.  ;)

**Comment by brianlagunas** [comment]
@dvorn this is not needed. It complicated the code base, and shows bad IoC guidance.  By default, the Prism region navigation framework reuses view/viewmodel instance unless you set the RegionMemberLifetime to false.

**Comment by dvorn** [comment]
@brianlagunas  Completed

**Pull request merged**